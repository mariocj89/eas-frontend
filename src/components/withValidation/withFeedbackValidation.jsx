import React from 'react';
import PropTypes from 'prop-types';

const withFeedbackValidation = WrappedComponent => {
  const WithFeedbackValidation = (props, context) => {
    const error = context.getFormError();
    return error ? <WrappedComponent error={error} /> : null;
  };

  WithFeedbackValidation.contextTypes = {
    getFormError: PropTypes.func.isRequired,
  };

  return WithFeedbackValidation;
};

export default withFeedbackValidation;
