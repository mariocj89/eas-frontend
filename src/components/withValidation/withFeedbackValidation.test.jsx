import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import withFeedbackValidation from './withFeedbackValidation';

const ValidationFeedback = withFeedbackValidation(() => <div />);

describe('withFeedbackValidation', () => {
  it('Should render the inner component correctly when has errors', () => {
    const wrapper = shallow(<ValidationFeedback />, {
      context: { getFormError: () => 'There is an error' },
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Should not render the inner component when doesn't have errors", () => {
    const wrapper = shallow(<ValidationFeedback />, { context: { getFormError: () => undefined } });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
