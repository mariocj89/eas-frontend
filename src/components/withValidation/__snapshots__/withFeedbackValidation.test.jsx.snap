// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withFeedbackValidation Should not render the inner component when doesn't have errors 1`] = `""`;

exports[`withFeedbackValidation Should render the inner component correctly when has errors 1`] = `
<Component
  error="There is an error"
/>
`;
